ksmt-symfpu/src/test/java/io/ksmt/symfpu/operations/FpToBvTransformerTest.kt

```diff
@@ -0,0 +1,689 @@
+            println("exprToTrans: ${unpackedString(baseExpr, model)}")
+            for ((name, expr) in printVars) {
+                val ufp = transformer.mapFpToUnpackedFp[expr.decl]
+                val evalUnpacked = unpackedString(ufp ?: expr, model)
+                println("$name :: $evalUnpacked")
+            }
+        }
+        assertNotEquals(KSolverStatus.SAT, status)
+    }
+
+
+    @EnabledIfEnvironmentVariable(
+        named = "runLongSymFPUTests",
+        matches = "true",
+    )
+    @Execution(ExecutionMode.CONCURRENT)
+    @Test
+    fun testFpToBvRoundToIntegralExpr() = with(createContext()) {
+        val a by mkFp32Sort()
+        val roundingModes = KFpRoundingMode.values()
+
+        roundingModes.forEach {
+            testFpExpr(
+                mkFpRoundToIntegralExpr(mkFpRoundingModeExpr(it), a),
+                mapOf("a" to a),
+            )
+        }
+    }
+
+
+    private fun KContext.unpackedString(value: KExpr<*>, model: KModel) = if (value.sort is KFpSort) {
+        val sb = StringBuilder()
+        val fpExpr: KExpr<KFpSort> by lazy { value.uncheckedCast() }
+        val ufp = if (value is UnpackedFp<*>) value else unpack(fpExpr.sort, mkFpToIEEEBvExpr(fpExpr.cast()), true)
+        val fpValue = model.eval(ufp.toFp()) as KFpValue
+        with(ufp) {
+            sb.append("uFP sign ")
+            model.eval(sign).print(sb)
+            sb.append(" ")
+            model.eval(unbiasedExponent).print(sb)
+            sb.append(" ")
+            model.eval(normalizedSignificand).print(sb)
+
+            //nan, inf, zero
+            sb.append(" nan=")
+            model.eval(isNaN).print(sb)
+            sb.append(" inf=")
+            model.eval(isInf).print(sb)
+            sb.append(" isZero=")
+            model.eval(isZero).print(sb)
+
+
+            sb.append("\ntoFp: ")
+            model.eval(toFp()).print(sb)
+
+            val packedFloat = packToBv(this)
+            val pWidth = packedFloat.sort.sizeBits.toInt()
+            val exWidth = sort.exponentBits.toInt()
+
+            // Extract
+            val packedSignificand = mkBvExtractExpr(pWidth - exWidth - 2, 0, packedFloat)
+            val packedExponent = mkBvExtractExpr(pWidth - 2, pWidth - exWidth - 1, packedFloat)
+            val sign = bvToBool(mkBvExtractExpr(pWidth - 1, pWidth - 1, packedFloat))
+            sb.append("\nFP sign ")
+            model.eval(sign).print(sb)
+            sb.append(" ")
+            model.eval(packedExponent).print(sb)
+            sb.append(" ")
+            model.eval(packedSignificand).print(sb)
+            sb.append("\nbv: ")
+            model.eval(packedFloat).print(sb)
+            sb.append(" \nactually ${(fpValue as? KFp32Value)?.value ?: (fpValue as? KFp16Value)?.value ?: (fpValue as? KFp128Value)}))}}")
+            sb.toString()
+        }
+    } else {
+        "${model.eval(value)}"
+    }
+
+    companion object {
```

Comment:
It is better to move these tests to the `ksmt-test` package and reuse infrastructure